test/json_client_test.rb

```diff
@@ -85,6 +85,17 @@ def test_should_fetch_and_parse_json_into_response
     assert_equal GdsApi::Response, @client.get_json(url).class
   end
 
+  def test_should_include_body_for_http_error
+    url = "http://some.endpoint/some.json"
+    stub_request(:get, url).to_return(body: "response body goes here", status: 404)
+    begin
+      @client.get_json(url)
+      assert_equal false, true
+    rescue GdsApi::HTTPErrorResponse => e
+      assert_equal "response body goes here", e.http_body
+    end
+  end
+
```

Comment:
This is a bit confusing. Can't we stub it to raise an error instead of relying on a failing assertion and a rescue block?

Reply:
It is raising an error, the `assert_equal true, false` bit is so that the test fails if the exception *isn't* raised

Reply:
Aha, `assert_raises` returns the error, I can simplify this: https://www.rubydoc.info/gems/minitest/Minitest%2FAssertions:assert_raises

Reply:
Much clearer, thank you